Sources/BrightroomEngine/Engine/ImageTool.swift

```diff
@@ -165,6 +266,21 @@ public enum ImageTool {
     return scaledImage
   }
 
+  public static func makeResizedCGImage(
+    from sourceImage: CGImage,
+    maxPixelSizeHint: CGFloat
+  ) -> CGImage? {
+    let maxPixelSize: CGFloat = {
+      let largestSide = max(sourceImage.size.width, sourceImage.size.height)
+      let smallestSide = min(sourceImage.size.width, sourceImage.size.height)
+      guard smallestSide >= maxPixelSizeHint else {
+        return largestSide
+      }
+      return largestSide * maxPixelSizeHint / smallestSide
+    }()
+    return makeResizedCGImage(from: sourceImage, maxPixelSize: maxPixelSize)
+  }
```

Comment:
thank you for fixing it.
Should we make the original one internal method? is it okay to keep it?

Reply:
which means, integrating this operation and below primitive operation.

Reply:
I thought it's ok to have both; since they do different things. If an user is confident that his image can use the `maxPixelSize` version it's ok to use it.